src/main/scala/WeaviateDataWriterFactory.scala

```diff
@@ -1,11 +1,17 @@
 package io.weaviate.spark
 
 import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.connector.write.streaming.StreamingDataWriterFactory
 import org.apache.spark.sql.connector.write.{DataWriter, DataWriterFactory}
 import org.apache.spark.sql.types.StructType
 
-case class WeaviateDataWriterFactory(weaviateOptions: WeaviateOptions, schema: StructType) extends DataWriterFactory with Serializable {
+case class WeaviateDataWriterFactory(weaviateOptions: WeaviateOptions, schema: StructType)
+  extends DataWriterFactory with StreamingDataWriterFactory with Serializable {
   override def createWriter(partitionId: Int, taskId: Long): DataWriter[InternalRow] = {
     WeaviateDataWriter(weaviateOptions, schema)
   }
+
+  override def createWriter(partitionId: Int, taskId: Long, epochId: Long): DataWriter[InternalRow] = {
```

Comment:
it is intended that you don't take into account `epochId` parameter?

Reply:
bc you're not passing it to `WeaviateDataWriter`

Reply:
this initial version of streaming support doesn't utilize epochId so it's intended. Just like we aren't using taskId either. Right now the streaming support re-uses the batch based datawriter and will utilize batching to load the stream data into Weaviate